### PR TITLE
Remove btn, it overwrites the other btn classes

### PR DESCRIPTION
**Pull request by pr33thi**
Interrupt ticket: https://dimagi-dev.atlassian.net/secure/RapidBoard.jspa?rapidView=5&projectKey=HI&modal=detail&selectedIssue=HI-375&quickFilter=3
Follow up to this PR: https://github.com/dimagi/commcare-hq/pull/23238

Turns out adding `btn` here overwrites btn-primary and btn-default, and makes the buttons look the same whether they're clicked or not (my bad, should have tested more thoroughly post-review). You mentioned that btn is what sets the formatting, but it seems to be formatted the same if I remove it. I guess that somewhere is bootstrap.css it defines btn-<whatever> as a kind of btn, does that seem likely?

@orangejenny 
@proteusvacuum

### CODE REVIEW

**Review by orangejenny — CHANGES_REQUESTED** [review]
TL;DR: This works but in a kind of roundabout way. Putting back the `.btn` class and adding `data-toggle="button"` to the button fixes it in a more intentional way.

Bootstrap applies javascript behavior to buttons. This is controlled by `bower_components/bootstrap/js/button.js`*. If you open that locally there's this code near the bottom, around line 110:
```
  $(document)
    .on('click.bs.button.data-api', '[data-toggle^="button"]', function (e) {
      var $btn = $(e.target).closest('.btn')
      Plugin.call($btn, 'toggle')
      ...
```
Every time a <button> element is clicked, this code runs. If the button has the `btn` class, it calls that `toggle` method on the button. `toggle` is defined around line 50, and at the end of it is this clause:
```
    } else {
      this.$element.attr('aria-pressed', !this.$element.hasClass('active'))
      this.$element.toggleClass('active')
    }
```
So what's happening is that you click the button, the `button_group.js` code runs and adds the `active` class, and then this bootstrap code runs and immediately removes the class, so it looks like nothing happened.

Bootstrap supports the kind of toggling behavior we want here: https://getbootstrap.com/docs/3.4/javascript/#buttons-single-toggle Using that would be more semantic.

> You mentioned that btn is what sets the formatting, but it seems to be formatted the same if I remove it. I guess that somewhere is bootstrap.css it defines btn- as a kind of btn, does that seem likely?

I see that removing `btn` doesn't seem to affect the styles and was surprised by that - you're always supposed to include `btn` when you use the `btn-foo` classes. It looks like we have a bunch of custom styles that are applied to both `.btn-group-separated > button` and `.btn-group-separated > .btn`, so the remaining differences are rules like `position: relative` that doesn't make a visible difference here.

\* This is the single-widget, unminified version. This code is also included in `bower_components/bootstrap/dist/js/bootstrap.min.js` which what's actually included in HQ: there's a script tag in [core_libraries.html](https://github.com/dimagi/commcare-hq/blob/d59f7435d621476c117fd22e9c4774cdb2ef6c5a/corehq/apps/hqwebapp/templates/hqwebapp/includes/core_libraries.html#L55) which is included by the [javascript_libraries](https://github.com/dimagi/commcare-hq/blob/d59f7435d621476c117fd22e9c4774cdb2ef6c5a/corehq/apps/hqwebapp/templatetags/hq_shared_tags.py#L737-L738) tag used in [hqwebapp/base.html](https://github.com/dimagi/commcare-hq/blob/d59f7435d621476c117fd22e9c4774cdb2ef6c5a/corehq/apps/hqwebapp/templates/hqwebapp/base.html#L147).

**Comment by pr33thi** [comment]
Thanks for the detailed explanation, it is appreciated.
https://github.com/dimagi/commcare-hq/pull/23341/commits/7eba278de747c92b936b066ea760d322d6fefa6b
The behavior around `active` getting removed as soon as it's added makes sense. And having a bunch of custom styles applied to the different button group makes sense, but how did you go about finding the css for these things? Is it just a matter of grepping for the classname or is there some organization system that I'm not aware of?

**Comment by orangejenny** [comment]
I usually grep for class names. As you've likely found, you can click on the css file names in the web dev styles panel, but that takes you to the compiled css, not the original less source, so it's not that helpful. It does at least help distinguish what's coming from HQ (style.randomness.css) versus bootstrap (bootstrap.random.css).

Style organization: Most of our styling code, of which maybe half customizes bootstrap classes and another half adds new classes, is in [hqwebapp/less/_hq](https://github.com/dimagi/commcare-hq/tree/master/corehq/apps/hqwebapp/static/hqwebapp/less/_hq/), which has a lot of files but they're well-named and most of them are short. Beyond that, web apps and app manager have additional less files, because they each have their own look and feel.

**Review by orangejenny — APPROVED** [review]
